pkg/volume/volume_linux.go

```diff
@@ -40,27 +41,96 @@ const (
 // SetVolumeOwnership modifies the given volume to be owned by
 // fsGroup, and sets SetGid so that newly created files are owned by
 // fsGroup. If fsGroup is nil nothing is done.
+var (
+	recorder record.EventRecorder
+	pod      *v1.Pod
```

Comment:
global pod variable?

Reply:
I am not sure how this will work. Can you explain what is the idea of these package/global variables?

Reply:
+1 these should probably be local & part of the goroutine closure.